tests/unit/providers/test_aws_provider.py

```
@@ -130,6 +130,103 @@ def test_no_api_key_skips_bearer_token_setup(monkeypatch: pytest.MonkeyPatch) ->
+    assert mock_session_cls.call_args_list[1].kwargs == expected_session_kwargs
+
+
+def test_profile_name_and_botocore_session_go_to_session_not_client() -> None:
+    """Test that profile_name/botocore_session reach boto3.Session but never Session.client().
+
+    boto3.Session() accepts `profile_name`/`botocore_session`, but `Session.client()` does not;
+    if they leaked into the `.client()` call, real boto3 would raise a TypeError.
+    """
+    fake_botocore_session = Mock()
+
+    def session_factory(**kwargs: Any) -> Mock:
```

Ruff (0.16.1)

[warning] Unused function argument: `kwargs`

(ARG001)